lib/engine/game/g_1846.rb

```diff
@@ -227,17 +227,23 @@ def action_processed(action)
       end
 
       def close_corporation(corporation)
+        @log << "#{corporation.name} closes"
+
         hexes.each do |hex|
           hex.tile.cities.each do |city|
-            next unless city.tokened_by?(corporation)
+            next unless city.tokened_by?(corporation) || city.reserved_by?(corporation)
 
             city.tokens.map! { |token| token&.corporation == corporation ? nil : token }
-            city.reservations.delete(corporation.name)
+            city.reservations.delete(corporation)
           end
         end
 
         corporation.spend(corporation.cash, @bank)
-        @log << "#{corporation.name} closes"
+        corporation.trains.each { |t| t.buyable = false }
+        if corporation.companies.any?
+          @log << "#{corporation.name}'s companies close: #{corporation.companies.map(&:sym).join(', ')}"
```

Comment:
do you think this is necessary?

Reply:
Yes.
I think other events that close companies are logged (some close when used, or the phase change), and I find it useful to log when any entity leaves the game.